expression/expr_to_pb.go

```diff
@@ -188,16 +189,16 @@ func collationToProto(c string) int32 {
 }
 
 func protoToCollation(c int32) string {
-	v, ok := mysql.Collations[uint8(collate.RestoreCollationIDIfNeeded(c))]
-	if ok {
-		return v
+	if coll, err := charset.GetCollationByID(int(collate.RestoreCollationIDIfNeeded(c))); err == nil {
+		return coll.Name
 	}
 	logutil.BgLogger().Warn(
```

Comment:
log the err

Reply:
updated

Reply:
The err and the original log say the same thing and the original log have more detail.  It's better to use the original code.